### PR TITLE
Allow test-runner to filter test groups by tag

### PR DESCRIPTION
**Pull request by dinatale2**
### Description

Enable test-runner to accept a list of tags to identify
which test groups the user wishes to run.

### Motivation and Context

I would like to use these tags to generate different classes of test runs in the future.
For example, a "trivial" test run or a "fast" test run could be useful for fuzz testing.
I would also like to allow buildbot to use the tags during zfs-test execution and allow
developers to specify tags to run in their commit messages.

### How Has This Been Tested?

Locally in a VM.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux code style requirements.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain `Signed-off-by`.
- [ ] Change has been approved by a ZFS on Linux member.

### CODE REVIEW

**Comment by dinatale2** [comment]
@jwk404 Would you mind providing input on this PR?

**Comment by gmelikov** [comment]
Thanks! Now we can easily add travis.yml from #5786!

**Comment by dinatale2** [comment]
Looks like I broke `zfs-tests.sh`. Will look into it later.

File: tests/runfiles/linux.run [hunk 1]
```diff
@@ -260,17 +303,21 @@ tests = ['zpool_events_clear', 'zpool_events_cliargs', 'zpool_events_follow',
 [tests/functional/cli_root/zpool_expand]
```
Comment (behlendorf) [inline tests/runfiles/linux.run]: Looks like you missed tagging the events tests.

File: tests/runfiles/linux.run [hunk 2]
```diff
@@ -306,30 +358,36 @@ tests = ['zpool_reopen_001_pos', 'zpool_reopen_002_pos',
```
Comment (behlendorf) [inline tests/runfiles/linux.run]: Same, missing tags for zpool_reopen.

File: scripts/zfs-tests.sh [hunk 3]
```diff
@@ -543,8 +549,13 @@ export __ZFS_POOL_EXCLUDE
 export TESTFAIL_CALLBACKS
 export PATH=$STF_PATH
 
-msg "${TEST_RUNNER} ${QUIET} -c ${RUNFILE} -i ${STF_SUITE}"
-${TEST_RUNNER} ${QUIET} -c "${RUNFILE}" -i "${STF_SUITE}"
+TAGSARG=
+if [[ -n $TAGS ]]; then
+	TAGSARG="-T $TAGS"
+fi
```
Comment (behlendorf) [inline scripts/zfs-tests.sh]: You could simplify this by having `TAGS="functional"` by default.  Then you can unconditionally pass `-T $TAGS` to test-runner to remove the need for `TAGSARG`.

File: tests/runfiles/linux.run [hunk 4]
```diff
@@ -8,7 +8,6 @@
 # source.  A copy of the CDDL is also available via the Internet at
 # http://www.illumos.org/license/CDDL.
 #
-
```
Comment (behlendorf) [inline tests/runfiles/linux.run]: nit: let's keep this line of white space

**Comment by dinatale2** [comment]
@jwk404 `zfs-test.sh` is no longer failing due to recent changes to the script.

**Comment by dinatale2** [comment]
I went ahead and updated the PR to include a `-I` flag to `test-runner` to control the number of times a test run is executed.

File: tests/runfiles/linux.run [hunk 5]
```diff
@@ -18,29 +18,35 @@ timeout = 600
 post_user = root
 post = cleanup
 outputdir = /var/tmp/test_results
+tags = []
```
Comment (behlendorf) [inline tests/runfiles/linux.run]: Why is a tags field  needed here at all?

Comment (dinatale2) [reply tests/runfiles/linux.run]: Refreshed. Defaulted to having the tag `functional` for any test group without tags specified.

**Pull request merged**